### PR TITLE
add clear search

### PR DESCRIPTION
**Pull request by rianfowler**
Deviated a little from the design- made the start over button aligned with the right side of the input. 

[ticket](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/11761)

screen shot 2018-08-02 at 13 20 25

---
screen shot 2018-08-02 at 13 20 33

### CODE REVIEW

File: src/applications/edu-benefits/components/SchoolSelectField.jsx [hunk 1]
```diff
@@ -133,19 +146,34 @@ export class SchoolSelectField extends React.Component {
...
+                type="text"
+                value={searchInputValue}/>
+              <div
+                className={clearSearchInfoClass}>
+                {showInstitutions && <span>
+                  {`${searchResultsCount} results for ${institutionQuery}`}
+                </span>}
+                <button
+                  onClick={this.handleStartOver}
+                  className="va-button-link">
```
Comment (rianfowler) [inline src/applications/edu-benefits/components/SchoolSelectField.jsx]: Did some reordering to fit the new flexbox layout. This button is the only new component.

File: src/applications/edu-benefits/sass/edu-benefits.scss [hunk 2]
```diff
@@ -42,13 +42,24 @@
...
+      flex-grow: 2;
+      flex-wrap: nowrap;
 
-    button {
+      .clear-search{
+        display: flex;
+        flex-direction: row;
+        justify-content: flex-end;
+        &.info {
+          justify-content: space-between;
```
Comment (rianfowler) [inline src/applications/edu-benefits/sass/edu-benefits.scss]: I really wanted to achieve this without class switching. Couldn't figure out how to get the Start Over link to be right justified when it was the only element in this container. 
screen shot 2018-08-02 at 13 20 33

File: src/applications/edu-benefits/components/SchoolSelectField.jsx [hunk 3]
```diff
@@ -133,19 +146,34 @@ export class SchoolSelectField extends React.Component {
...
+                  Start Over
+                </button>
+              </div>
+            </div>
+            <div
+              className="search-schools">
```
Comment (cehsu) [inline src/applications/edu-benefits/components/SchoolSelectField.jsx]: Should we name this "search-trigger" to match "search-input" above? Also, perhaps we want to keep this name generic in order to make it easier to convert into a common/shared component later.

**Review by cehsu — APPROVED** [review]

File: src/applications/edu-benefits/components/SchoolSelectField.jsx (hunk 3 above)
Comment (rianfowler) [reply src/applications/edu-benefits/components/SchoolSelectField.jsx]: I like that idea. I can do some renaming of the props and keep the domain specific items in the redux and react-redux stuff.

**Pull request merged**